src/backend/access/nbtree/nbtsearch.c

```diff
@@ -2421,6 +2498,24 @@ _bt_endpoint(IndexScanDesc scan, ScanDirection dir)
 		return false;
 	}
 
+	so->prefetch_maximum = IsCatalogRelation(rel)
+		? effective_io_concurrency
+		: get_tablespace_io_concurrency(rel->rd_rel->reltablespace);
+	/* Neon: we currently do not use prefetch for parallel index scan
+	 * because prefetch is efficient only if prefetched blocks are accessed by the same worker which issued prefetch request,
+	 * logic of splitting pages between parallel  worked in index scan doesn't allow to satisfy this requirement.
+	 * Also prefetch of leaf pages can be efficient only for index-only scans because otherwise
+	 * time is mostly spent in reading heap pages.
```

Comment:
Yes, it is possible to implement prefetch for index scan in this way.
But it seems to e completely different approach: instead of traversing parent pages, we just need to prefetch all TIDs from leaf page. I am not sure if it is reasonable to try to combine this two approaches. Average number of keys on a pages is about 100 which is comparable with default prefetch distance (`effective_io_concurrency`).

Reply:
> Did you index more (or wider) columns?

No, my estimation was almost the same: 4-10 ley length and ~1/2 page filling.
Sorry, but I do not see any contradiction with with my statement.
When I wrote "about 100" I do not mean  that 100 is precise average value. 174 IMHO is also "about 100":)
What I mean that this number is larger (or at least comparable) with prefetch distance, i.e.
it is enough to  prefetch TIDs on leaf page rather then leave pages themselves. Certainly prefetching leave pages in this case will give some positive effect, but I expect that it will be negligible.

Also I have a feeling (it is just a feeling I can not prove it by some real use cases) that Postgres optimizer very rarely choose index scan (not index-only scan) for the whole relation or at least large subset of relation = where prefetching may be useful. It is mostly because of quite larger (4 times by default) difference between random and sequential page access cost.
Queries with large ranges can be quite popular for timeseries, but here is makes sense to create covering index to make  it possible to use index-only scan.